plugin-api/src/main/java/org/sonar/api/server/rule/internal/ImpactMapper.java

```diff
@@ -0,0 +1,116 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.server.rule.internal;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import org.sonar.api.issue.impact.Severity;
+import org.sonar.api.issue.impact.SoftwareQuality;
+import org.sonar.api.rules.RuleType;
+
+import static org.sonar.api.rule.Severity.BLOCKER;
+import static org.sonar.api.rule.Severity.CRITICAL;
+import static org.sonar.api.rule.Severity.INFO;
+import static org.sonar.api.rule.Severity.MAJOR;
+import static org.sonar.api.rule.Severity.MINOR;
+
+/**
+ * @deprecated since 10.1 This is only used for mapping deprecated types and severities until they are removed
+ */
+@Deprecated(since = "10.1")
+public class ImpactMapper {
+
+  //TODO maybe add a test to make sure all the software qualities are added
+  static final List<SoftwareQuality> ORDERED_SOFTWARE_QUALITIES = List.of(SoftwareQuality.MAINTAINABILITY,
+    SoftwareQuality.RELIABILITY, SoftwareQuality.SECURITY);
+
+  private ImpactMapper() {
+    // This class is designed to be static
+  }
+
+  public static SoftwareQuality convertToSoftwareQuality(RuleType ruleType) {
+    switch (ruleType) {
+      case CODE_SMELL:
+        return SoftwareQuality.MAINTAINABILITY;
+      case BUG:
+        return SoftwareQuality.RELIABILITY;
+      case VULNERABILITY:
+        return SoftwareQuality.SECURITY;
+      case SECURITY_HOTSPOT:
+        throw new IllegalStateException("Can not map Security Hotspot to Software Quality");
+      default:
+        throw new IllegalStateException("Unknown rule type");
+    }
+  }
+
+  public static RuleType convertToRuleType(SoftwareQuality softwareQuality) {
+    switch (softwareQuality) {
+      case MAINTAINABILITY:
+        return RuleType.CODE_SMELL;
+      case RELIABILITY:
+        return RuleType.BUG;
+      case SECURITY:
+        return RuleType.VULNERABILITY;
+      default:
+        throw new IllegalStateException("Unknown software quality");
+    }
+  }
+
+  public static String convertToDeprecatedSeverity(Severity severity) {
+    switch (severity) {
+      case HIGH:
+        return CRITICAL;
+      case MEDIUM:
+        return MAJOR;
+      case LOW:
+        return MINOR;
+      default:
+        throw new IllegalStateException("This severity value " + severity + " is illegal.");
+    }
+  }
+
+  public static org.sonar.api.issue.impact.Severity convertToSeverity(String deprecatedSeverity) {
```

Comment:
naming suggestion: `convertToImpactSeverity` to be more clear on what the methods do

Reply:
done